package-examples/function-export/README.md

```diff
@@ -0,0 +1,103 @@
+Putting a validation function  into your package allows you to give package 
+consumers guidance on what acceptable configuration is while giving them 
+the freedom to customize.
+
+## Steps
+
+This is a simple workflow where you can download, configure, render,
+validate and apply the package:
+
+1. [Fetch the package](#fetch-the-package)
+2. [View the package contents](#view-the-package-contents)
+3. [Configure functions](#configure-functions)
+4. [Render the declared values](#render-the-declared-values)
+5. [Apply the package](#apply-the-package)
+
+### Fetch the package
+
+Get the example package on to local using `kpt pkg get`
+
+  $ kpt pkg get https://github.com/GoogleContainerTools/kpt.git/package-examples/function-export
+
+    fetching package /package-examples/function-export from https://github.com/GoogleContainerTools/kpt to function-export
+
+### View the package contents
+
+List the package contents in a tree structure.
+
+  $ kpt pkg tree function-export/
+
+    PKG: function-export
+    ├── [Kptfile]  Kptfile kpt-function-export-example
+    └── resources
+        ├── [resources.yaml]  Namespace development
+        ├── [resources.yaml]  Deployment development/nginx-deployment
+        └── constraints
+            ├── [deployment-must-have-owner.yaml]  K8sRequiredLabels deployment-must-have-owner
+            └── [requiredlabels.yaml]  ConstraintTemplate k8srequiredlabels
+
+### Configure functions
+
+The package contains a function pipeline in the `Kptfile` which has
+one `apply-setters` and `gatekeeper-validate` functions.  
+The `apply-setters` function allows you to set a simple value throughout the 
+package configuration.  In this case it's namespace label.  The
+`gatekeeper-validate` function allows you to use gatekeeper for checks on
+the configuration.
+
+  pipeline:
+    mutators:
+      - image: gcr.io/kpt-functions/label-namespace
+        configMap:
+          label_name: color
+          label_value: blue      
+    validators:
+      - image: gcr.io/kpt-functions/gatekeeper-validate
+
+
+
+### Render the declared values
+
+Render the changes in the hydration pipeline by using `kpt fn render` command:
+
+  $ kpt fn render function-export/
+
+    package "function-export": running function "gcr.io/kpt-functions/label-namespace": SUCCESS
+    package "function-export": running function "gcr.io/kpt-functions/gatekeeper-validate": SUCCESS
+    package "function-export": rendered successfully
+
+
+If you remove the owner label from `resources.yaml` and re-run the rendering
+you should see an error:
+
+
+  $ kpt fn render function-export/
+
+    kpt fn render function-export/ 
+    package "function-export": running function "gcr.io/kpt-functions/label-namespace": SUCCESS
+    package "function-export": running function "gcr.io/kpt-functions/gatekeeper-validate": FAILED
+    fn.render: pkg function-export:
+            pkg.render:
+            pipeline.run: Error: Found 1 violations:
```

Comment:
Error reporting in `fn render` still needs work :) but better than before.

Reply:
Indeed.  This is a step in the right direction.  I am sure users will give us good guidance on what else we can do to help.